src/lighteval/models/base_model.py

```diff
@@ -753,9 +900,20 @@ def prepare_batch_logprob(
             padded=padded,
         )
 
-    def pad_and_gather(self, output_tensor: torch.Tensor) -> torch.Tensor:
-        """Gather together tensors of (possibly) various size spread on separate GPUs (first exchange the lengths and then pad and gather)"""
-        # Create a tensor of size batch_size, [output_length] * batch_size, for each each process
+    def pad_and_gather(self, output_tensor: torch.Tensor, drop_last_samples: bool = True) -> torch.Tensor:
+        """
+        Pads the `output_tensor` to the maximum length and gathers the lengths across processes.
+
+        Args:
+            output_tensor (torch.Tensor): The output tensor to be padded.
+            drop_last_samples (bool, optional): Whether to drop the last samples during gathering.
+            Last samples are dropped when the number of samples is not divisible by the number of processes.
```

Comment:
In which cases would we need to drop samples? Won't it change the results? Why do we allow this?

Reply:
Thanks for the explanation!
I don't think we want to drop samples without checking if there is a max_samples passed, and without warning. Shouldn't be too much of a problem as iirc there's a function checking the number of items before and after processing, but we should make sure there's no accident

Reply:
It does not have to be max_samples, we just need the number of samples to not be devisible by the DP.
Also, its safe to do, the function `gather_for_metrics` does it automatically. It's an accelerate mechanism.

see: https://huggingface.co/docs/accelerate/en/package_reference/accelerator#accelerate.Accelerator.gather

Reply:
I added a mechanism to NOT drop samples that have been added by accelerate.

Reply:
Thanks for the ref, I'll look more in depth into this